core/src/main/java/io/questdb/cairo/ColumnPurgeJob.java

```diff
@@ -232,7 +232,9 @@ private void processTableRecords(CairoEngine engine) {
                     long ts = rec.getTimestamp(0);
                     if (ts != lastTs || taskRun == null) {
                         if (taskRun != null) {
-                            columnPurgeOperator.purgeExclusive(taskRun);
+                            if (!taskRun.isEmpty()) {
+                                columnPurgeOperator.purgeExclusive(taskRun);
```

Comment:
One more scenario I am not sure about: 

1. You just successfully purged a table. 
2. In the next iteration, you pick a table which happens to be dropped. Thus the guard on line no. 253 will short-circuit the loop. 
3. The dropped table was the last record -> `records.hasNext()` returns false and you exit the loop. 
4. At this point the `taskRun` is not null and it's set to the table already purged in step no. 1.
5. So the line no. 276 will purge it again!

In other words: If the last table is dropped then the last non-dropped table will be purged twice. I am not sure if that may cause a problem (probably not), just highlighting it for your consideration.

Reply:
I changed the code to avoid double runs, thanks